concrete-test/src/raw/generation.rs

```diff
@@ -0,0 +1,98 @@
+//! A module containing sampling entry points for raw integers
+use concrete_commons::dispersion::{DispersionParameter, Variance};
+use concrete_commons::numeric::{CastInto, UnsignedInteger};
+use concrete_core::backends::core::private::math::random::RandomGenerator;
+use kolmogorov_smirnov;
+
+/// A trait to generate raw unsigned integer values.
+pub trait RawUnsignedIntegers: UnsignedInteger + CastInto<f64> {
+    fn one() -> Self;
+    fn one_vec(size: usize) -> Vec<Self>;
+    fn zero() -> Self;
+    fn zero_vec(size: usize) -> Vec<Self>;
+    fn uniform() -> Self;
+    fn uniform_vec(size: usize) -> Vec<Self>;
+    fn uniform_n_msb(n: usize) -> Self;
+    fn uniform_n_msb_vec(n: usize, size: usize) -> Vec<Self>;
+    fn uniform_weight() -> Self;
+}
+
+impl RawUnsignedIntegers for u32 {
+    fn one() -> Self {
+        1u32
+    }
+    fn one_vec(size: usize) -> Vec<Self> {
+        vec![1u32; size]
+    }
+    fn zero() -> Self {
+        0u32
+    }
+    fn zero_vec(size: usize) -> Vec<Self> {
+        vec![0u32; size]
+    }
+    fn uniform() -> Self {
+        let mut generator = RandomGenerator::new(None);
+        generator.random_uniform()
+    }
+    fn uniform_vec(size: usize) -> Vec<Self> {
+        let mut generator = RandomGenerator::new(None);
+        generator.random_uniform_tensor(size).into_container()
+    }
+    fn uniform_n_msb(n: usize) -> Self {
+        let mut generator = RandomGenerator::new(None);
+        generator.random_uniform_n_msb(n)
+    }
+    fn uniform_n_msb_vec(n: usize, size: usize) -> Vec<Self> {
+        let mut generator = RandomGenerator::new(None);
+        generator
+            .random_uniform_n_msb_tensor(size, n)
+            .into_container()
+    }
+    fn uniform_weight() -> Self {
+        let val: u32 = Self::uniform();
+        let val = val % 1024u32;
+        let val: i32 = val as i32;
+        let val = val - 512i32;
+        val as u32
+    }
```

Comment:
Hello @blowfish880 !

This function comes from this test:
https://github.com/zama-ai/concrete/blob/f5a1f21c108a1a514f3d27883f0095d671d6a7d0/concrete-core/src/backends/core/private/crypto/lwe/tests.rs#L363-L367

I believe it samples a random value between -512 and 512 (represented as signed integers). I don't really know why this value was picked in the first place to be honest.

Maybe a function in this spirit would be better ? :
```rust
fn uniform_zero_centered(half_size:usize) -> Self {
        let val: u64 = Self::uniform();
        let val = val % (half_size*2 as u64);
        let val: i64 = val as i64;
        let val = val - (half_size as i64);
        val as u64
}
```

Reply:
Yes, that name seems definitely better to me, but it may suggest that the returned value is signed. But maybe I'm overthinking this.

Reply:
On this side, I would say that since the function is returning `Self` which is `RawUnsignedInteger`, it is not too problematic. I used the `uniform_zero_centered` name !